libraries/AP_Generator/AP_Generator_RichenPower.cpp

```diff
@@ -200,26 +254,15 @@ void AP_Generator_RichenPower::update_runstate()
 
 void AP_Generator_RichenPower::update_servo_channel(void)
 {
+    if (AP::arming().is_armed()) {
+        return;
+    }
     const uint32_t now = AP_HAL::millis();
     if (now - _last_servo_channel_check < 1000) {
         return;
     }
     _last_servo_channel_check = now;
-    SRV_Channel *control = SRV_Channels::get_channel_for(SRV_Channel::k_richenpower_control);
-    if (control == nullptr) {
-        if (_servo_channel != nullptr) {
-            gcs().send_text(MAV_SEVERITY_INFO, "RichenPower: no control channel");
-            _servo_channel = nullptr;
-        }
-        return;
-    }
-    if (_servo_channel != nullptr &&
-        _servo_channel->get_function() == SRV_Channel::k_richenpower_control) {
-        // note that _servo_channel could actually be == control here
-        return;
-    }
-    // gcs().send_text(MAV_SEVERITY_INFO, "RP: using control channel");
-    _servo_channel = control;
+    _servo_channel = SRV_Channels::get_channel_for(SRV_Channel::k_richenpower_control);
```

Comment:
Can we please just remove all this? Se shouldn't be calling `set_output_pwm` on a signle channel, we should just be using the SRV_Channel method `set_output_pwm(SRV_Channel::Aux_servo_function_t function, uint16_t value)` thus all the calls would just tweak to be `SRV_Channels::set_output_pwm(SRV_Channel::k_richenpower_control, value)` which is lifting the mapping/tracking of the state out of the library nicely.

I can see doing the lookup once as part of the prearm check just to see that a channel exists, but I'm not sure it's worth the memory to keep the pointer to a single channel here.

Reply:
As a a side note, we probably should name the servo channel "k_generator_control" instead of being so specific that it's a richenpower generator.  If we had another backend which had a pwm output we would use the same output channel.

Reply:
@WickedShell sounds good.  I was thinking of the multi-generator case.  Probably going a bit far...

Reply:
@rmackay9 I did consider that.  I settled on this on the basis we could always rename it later.  I don't think the chances are high that we should reuse exactly the same PWM values to mean exactly the same thing on different generators, thus the specific naming.

Reply:
@peterbarker multi generator won't work anyways without a different option per channel, or putting the param into the library, so it doesn't really matter

Reply:
Moved to using `set_output_pwm`

Reply:
Thanks!